functions/module_metadata/os_blacklisted.pp

```diff
@@ -0,0 +1,98 @@
+# Returns whether or not the passed module is blacklisted per the module's
+# metadata.json.
+#
+# If a blacklist is passed, then it will return `false` if the OS is in the
+# blacklist and `true` otherwise.
+#
+# If the blacklist is not matched, the standard OS validation will be used.
```

Comment:
Did I miss that logic?

Reply:
Nah, that was a holdover from the 5th rewrite :-|